src/Engine/ProtoCore/Namespace/ElementResolver.cs

```diff
@@ -0,0 +1,68 @@
+﻿using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ProtoCore.Namespace
+{
+    /// <summary>
+    /// Responsible for resolving a partial class name to its fully resolved name
+    /// </summary>
+    public class ElementResolver
+    {
+        private Dictionary<string, KeyValuePair<string, string>> resolutionMap;
+
+        /// <summary>
+        /// Maintains a lookup table of partial class identifiers vs. 
+        /// fully qualified class identifier names and assembly name
+        /// </summary>
+        public Dictionary<string, KeyValuePair<string, string>> ResolutionMap
+        {
+            get { return resolutionMap; }
+        }
+
+        #region public constructors and methods
+
+        public ElementResolver()
+        {
+            resolutionMap = new Dictionary<string, KeyValuePair<string, string>>();
+        }
+
+        public ElementResolver(Dictionary<string, KeyValuePair<string, string>> namespaceLookupMap)
```

Comment:
I think the equivalent of `IEnumerable` should be used here (something like `IDictionary`?)